docs/GuidGuidRecommender.md

```diff
@@ -0,0 +1,144 @@
+we apply various [metrics](./GraphRepresentation.md#quality-and-health-metrics)
+to the coinstallation data, candidate lists, and recommendations,
+to evaluate the overall health of the system
+and the quality of recommendations.
+
+The recommender system is implemented by the [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10) class,
+which is instantiated by the [TaarLiteAppResource](../taar_lite/app/production.py#L44).
+
+Further technical details are provided on [this page](./GraphRepresentation.md).
+
+
+# Input data
+
+The primary data source used in generating recommendations
+is the file `guid_coinstallation.json`,
+which is generated by the ETL jobs described in the [README](../README.md).
+It contains coinstallation counts for each pair of add-ons
+belonging to a filtered list.
+
+Coinstallation counts are computed based on the Telemetry data
+submitted by a random sample of recently active Firefox profiles.
+For each pair of add-ons in the list,
+we count the number of profiles in which both add-ons are installed and enabled.
+
+The data is stored in "sparse matrix representation" as a nested object/dict,
+in which each add-on GUID maps to a further object
+mapping coinstalled GUIDS to their coinstallation count,
+dropping zero counts.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `raw_coinstall_dict`.
+
+Using this dataset as the basis for recommendations
+entails the following assumptions:
+
+- The set of add-ons appearing in the coinstallation dataset represent
+    the universe of known add-ons for the purpose of generating recommendations.
+- We are only dealing with add-ons that were coinstalled
+    with at least one other add-on on the list
+    (otherwise they would not appear in the file at all).
+- Profiles with only a single add-on installed do not contribute to the counts.
+
+__TODO:__ How are system add-ons handled? Add a note about this.
+
+In computing relevance scores, we also draw on an auxiliary dataset
+containing an overall ranking for each add-on in our list,
+read from the file `guid_install_ranking.json`.
+This is used to break ties when sorting candidates on relevance score,
+and in pruning out rare add-ons.
+It is currently computed from overall add-on installation counts
+obtained from AMO.
+
+This data is stored as an object/dict mapping add-on GUIDs to rank scores.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `tie_breaker_dict`.
+It is also used as the `ranking_dict` passed to the [`MinInstallPrune`](../taar_lite/recommenders/treatments.py#L36) treatment.
+
+
+# Generating the scored candidate lists
+
+The TAAR-lite recommender is built on the following underlying principle:
+_add-ons which tend to be installed together are similar or complimentary,
+and the number of coinstallations indicates the strength of this relationship._
```

Comment:
I would say they're much more likely to be complimentary than similar.

I somehow wanted this to be broader e.g. about the fact that it's super anonymous data, but can provide valuable add-on recommendations.....but I can't come up with anything succinct and I don't think i'ts' necessary.